### DISCUSSION TITLE
Allow lowercase variables and reserve future keywords

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2833] [Suggestions]
I've been pushing modern E2's syntax to get closer to typescript/javascript so learning E2 is easier and more educational for people wanting to use E2 as a step before moving on to a real world language

The syntax has gotten to a point where it's pretty much identical minus E2 having some extra features. Last things that need to be done are this, #2867 and #2771.
 
Before doing this, need to create a list of keywords to reserve for the future though, here's my proposed keywords

### Variables
`static` `super` ~~`self`~~ `true` `false` `mut` `var` `null` `undefined`

### Structures
`abstract` `class` `struct` `enum` `union` `extends` `implements` `instanceof` `new` `constructor` `type` `interface`

### Macros
`macro`

### Looping
`in` `of`

### Exceptions
`throw` `throws` `finally`

### Modules
`mod` `import` `namespace` `export` `public` `private` `pub` `priv` `package` `use`

### Generators / Async
`await` `async` ~~`generator` `gen` `coroutine`~~ (should be done implicitly) `yield`

### Misc
`match` `as` `typeof`

Feel free to comment below any other keywords you think should be reserved. More is better, we can always just remove them later on if it's agreed that we won't implement the actual functionality.

**Comment by bjcscat** [comment]
`client`, `server`, `sync`, `shared`

**Comment by vurvdev** [comment]
`using` `unknown` `dynamic`

**Comment by Denneisk** [comment]
Can't have `constructor` without `this` (Translation: please spare me from writing `This` instead)

**Reply by vurvdev** [reply]
Ah yeah always forget JS uses `this` instead of `self`.

Should remove `self` then.

(Ignoring the fact that there's a bunch of non-js keywords here)

**Comment by Denneisk** [comment]
#3201 will add a warning if any of these are used as an identifier. Hope the impl is alright :P
